From 0cac816ddadb5846f5ad5e63314ceeeccac5018f Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Thu, 30 Oct 2025 10:50:05 +0100
Subject: [PATCH] Nuke Knox HDM

UN1CA: this patch is not complete! Please check unica/mods/deknox/customize.sh for more info.
---
 .../com/samsung/android/knox/hdm/HdmManager.smali  | 268 +++------------------
 1 file changed, 30 insertions(+), 238 deletions(-)

diff --git a/smali_classes4/com/samsung/android/knox/hdm/HdmManager.smali b/smali_classes4/com/samsung/android/knox/hdm/HdmManager.smali
index ce2d6c2..9a551c5 100644
--- a/smali_classes4/com/samsung/android/knox/hdm/HdmManager.smali
+++ b/smali_classes4/com/samsung/android/knox/hdm/HdmManager.smali
@@ -35,7 +35,7 @@
 .end method
 
 .method public static getHdmVersion()Ljava/lang/String;
-    .locals 4
+    .locals 2
 
     sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
@@ -43,40 +43,7 @@
 
     invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
-    const-string v0, "ro.vendor.hdm.hdm_supported_subsystem"
-
-    const-string v1, "none"
-
-    invoke-static {v0, v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v2
-
-    invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result v2
-
-    if-nez v2, :cond_0
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    const-string v3, "2.0 - "
-
-    invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-static {v0, v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v0
-
-    return-object v0
-
-    :cond_0
-    const-string v0, "2.0"
+    const/4 v0, 0x0
 
     return-object v0
 .end method
@@ -84,43 +51,18 @@
 
 # virtual methods
 .method public generateNonce()Ljava/lang/String;
-    .locals 2
+    .locals 1
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "generateNonce() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling generateNonce() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0}, Lcom/samsung/android/knox/hdm/IHdmManager;->generateNonce(Lcom/samsung/android/knox/ContextInfo;)Ljava/lang/String;
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    move-result-object p0
+    const-string v0, "generateNonce() on HdmManager.java"
 
-    return-object p0
-
-    :cond_0
-    const-string p0, "Fail to call generateNonce() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -128,43 +70,18 @@
 .end method
 
 .method public getHdmId(Ljava/lang/String;)Ljava/lang/String;
-    .locals 2
+    .locals 0
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "getHdmId() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling getHdmId() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0, p1}, Lcom/samsung/android/knox/hdm/IHdmManager;->getHdmId(Lcom/samsung/android/knox/ContextInfo;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call getHdmId() using HDM Service on HdmManager.java"
+    const-string p1, "getHdmId() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -172,43 +89,18 @@
 .end method
 
 .method public getHdmIdB64()Ljava/lang/String;
-    .locals 2
+    .locals 1
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "getHdmIdB64() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling getHdmIdB64() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0}, Lcom/samsung/android/knox/hdm/IHdmManager;->getHdmIdB64(Lcom/samsung/android/knox/ContextInfo;)Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call getHdmIdB64() using HDM Service on HdmManager.java"
+    const-string v0, "getHdmIdB64() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -216,43 +108,18 @@
 .end method
 
 .method public getHdmPolicy(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-    .locals 2
+    .locals 0
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "getHdmPolicy() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling getHdmPolicy() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0, p1, p2}, Lcom/samsung/android/knox/hdm/IHdmManager;->getHdmPolicy(Lcom/samsung/android/knox/ContextInfo;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call getHdmPolicy() using HDM Service on HdmManager.java"
+    const-string p1, "getHdmPolicy() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -260,43 +127,18 @@
 .end method
 
 .method public getRevocationInfo()Ljava/lang/String;
-    .locals 2
+    .locals 1
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "getRevocationInfo() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling getRevocationInfo() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0}, Lcom/samsung/android/knox/hdm/IHdmManager;->getRevocationInfo(Lcom/samsung/android/knox/ContextInfo;)Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call getRevocationInfo() using HDM Service on HdmManager.java"
+    const-string v0, "getRevocationInfo() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -329,43 +171,18 @@
 .end method
 
 .method public isNFCBlockedByHDM()Z
-    .locals 2
+    .locals 1
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "isNFCBlockedByHDM() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling isNFCBlockedByHDM() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0}, Lcom/samsung/android/knox/hdm/IHdmManager;->isNFCBlockedByHDM(Lcom/samsung/android/knox/ContextInfo;)Z
-
-    move-result p0
-
-    return p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call isNFCBlockedByHDM() using HDM Service on HdmManager.java"
+    const-string v0, "isNFCBlockedByHDM() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
@@ -373,43 +190,18 @@
 .end method
 
 .method public isSwBlockEnabled()Z
-    .locals 2
+    .locals 1
     .annotation system Ldalvik/annotation/Throws;
         value = {
             Landroid/os/RemoteException;
         }
     .end annotation
 
-    sget-object v0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
-
-    const-string v1, "isSwBlockEnabled() on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/hdm/HdmManager;->getService()Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    move-result-object v1
-
-    if-eqz v1, :cond_0
-
-    const-string v1, "Calling isSwBlockEnabled() using HDM Service on HdmManager.java"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mService:Lcom/samsung/android/knox/hdm/IHdmManager;
-
-    iget-object p0, p0, Lcom/samsung/android/knox/hdm/HdmManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-interface {v0, p0}, Lcom/samsung/android/knox/hdm/IHdmManager;->isSwBlockEnabled(Lcom/samsung/android/knox/ContextInfo;)Z
-
-    move-result p0
-
-    return p0
+    sget-object p0, Lcom/samsung/android/knox/hdm/HdmManager;->TAG:Ljava/lang/String;
 
-    :cond_0
-    const-string p0, "Fail to call isSwBlockEnabled() using HDM Service on HdmManager.java"
+    const-string v0, "isSwBlockEnabled() on HdmManager.java"
 
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
+    invoke-static {p0, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
     const/4 p0, 0x0
 
-- 
2.51.1
